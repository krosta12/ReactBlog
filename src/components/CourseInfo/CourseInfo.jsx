import '../../App.css';

export default function CourseInfo(props) {
	return (
		<div className='Card'>
			<div className='Texts'>
				<h2>{props.post.title}</h2>
				{/* <Button2 text='back' onClick={alert('D')} /> какойто максимально тупой баг с копкой*/}
				<p>{props.post.description}</p>
			</div>
			<div className='Info'>
				<div className='InfoInner'>
					<div className='authors'>
						Authors:{' '}
						{props.post.authors.map((el) => {
							return <span>{el} </span>;
						})}
					</div>
					<p>Duration: {props.post.duration} Hours</p>
					<p>Created: {props.post.creationDate}</p>
				</div>
			</div>
		</div>
	);
}
