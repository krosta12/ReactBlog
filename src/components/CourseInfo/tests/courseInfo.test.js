import { render, screen } from '@testing-library/react';
import CourseInfo from '../CourseInfo';
import { BrowserRouter } from 'react-router-dom';
import '@testing-library/jest-dom/extend-expect'; //stack-overflow expect(...).toBeInTheDocument is not a function

describe('test case', () => {
	let post = {
		title: 'How to save...',
		creationDate: '24/06/2023',
		description: `Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to ma`,
		duration: '106.06',
		id: 'cc0c5455-2836-40bd-932c-e7d9e06de0b6',
		authors: ['AuthorNameNum1', 'AuthorNameNum2'],
	};

	const renderBlock = () =>
		render(
			<BrowserRouter>
				<CourseInfo post={post} />
			</BrowserRouter>
		);

	test('Title Test', () => {
		renderBlock();
		const linkElement = screen.getByText(post.title);
		expect(linkElement).toMatchSnapshot();
	});

	test('Description Test', () => {
		renderBlock();
		const linkElement = screen.getByText(post.description);
		expect(linkElement).toMatchSnapshot();
	});

	test('Duration Test', () => {
		renderBlock();
		const durationElement = screen.getByText(/Duration: \d+\.\d+ Hours/);
		expect(durationElement.textContent).toMatchSnapshot();
	});
	test('Authors List Test', () => {
		const author = renderBlock();
		expect(author).toMatchSnapshot();
	});
	test('Course Card Created Date Test', () => {});
});
