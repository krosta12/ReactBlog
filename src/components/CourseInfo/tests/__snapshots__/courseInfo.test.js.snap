// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test case Authors List Test 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="Card Main"
      >
        <button>
          back
        </button>
        <div
          class="Texts"
        >
          <h2>
            How to save...
          </h2>
          <p>
            Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to ma
          </p>
        </div>
        <div
          class="Info"
        >
          <div
            class="InfoInner infoInnerCard"
          >
            <div
              class="authors"
            >
              Authors:
               
              <span>
                AuthorNameNum1
                 
              </span>
              <span>
                AuthorNameNum2
                 
              </span>
            </div>
            <p>
              Duration: 
              106.06
               Hours
            </p>
            <p>
              Created: 
              24/06/2023
            </p>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="Card Main"
    >
      <button>
        back
      </button>
      <div
        class="Texts"
      >
        <h2>
          How to save...
        </h2>
        <p>
          Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to ma
        </p>
      </div>
      <div
        class="Info"
      >
        <div
          class="InfoInner infoInnerCard"
        >
          <div
            class="authors"
          >
            Authors:
             
            <span>
              AuthorNameNum1
               
            </span>
            <span>
              AuthorNameNum2
               
            </span>
          </div>
          <p>
            Duration: 
            106.06
             Hours
          </p>
          <p>
            Created: 
            24/06/2023
          </p>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`test case Description Test 1`] = `
<p>
  Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to ma
</p>
`;

exports[`test case Duration Test 1`] = `"Duration: 106.06 Hours"`;

exports[`test case Title Test 1`] = `
<h2>
  How to save...
</h2>
`;
